parquet_reader/hpat_parquet_reader.cpp

```diff
@@ -86,11 +86,13 @@ int64_t pq_get_size_single_file(std::shared_ptr<FileReader> arrow_reader, int64_
 int64_t
     pq_read_single_file(std::shared_ptr<FileReader> arrow_reader, int64_t column_idx, uint8_t* out_data, int out_dtype)
 {
-    std::shared_ptr<::arrow::Array> arr;
-    arrow_reader->ReadColumn(column_idx, &arr);
-    if (arr == NULL)
+    std::shared_ptr<::arrow::ChunkedArray> chunked_array;
+    arrow_reader->ReadColumn(column_idx, &chunked_array);
+    if (chunked_array == NULL)
         return 0;
 
+    auto arr = chunked_array->chunk(0);
```

Comment:
Does it meant that we always have one chunk after reading?

Reply:
It corresponds to [original implementation](https://github.com/apache/arrow/blob/9fb5c3d8f249482a5e3c0d33d7626c438e28c774/cpp/src/parquet/arrow/reader.cc#L775) of `ReadColumn(... <Array> ...). It gets `ChunkedArray` end returns only the first `Array` in it.
I think this code should be improved in the future, e.g. for using all arrays in ChunkedArray. But for porting reasons it is ok because it works as previously.